src/app/server/Mdns.cpp

```diff
@@ -93,30 +87,37 @@ CHIP_ERROR GetCommissionableInstanceName(char * buffer, size_t bufferLen)
 /// Set MDNS operational advertisement
 CHIP_ERROR AdvertiseOperational()
 {
-    uint64_t fabricId;
-
-    if (DeviceLayer::ConfigurationMgr().GetFabricId(fabricId) != CHIP_NO_ERROR)
+    for (const Transport::FabricInfo & fabricInfo : GetGlobalFabricTable())
     {
-        ChipLogError(Discovery, "Fabric ID not known. Using a default");
-        fabricId = 5544332211;
+        if (fabricInfo.IsInitialized())
+        {
+            uint8_t mac[8];
+
+            const auto advertiseParameters =
+                chip::Mdns::OperationalAdvertisingParameters()
+                    .SetPeerId(PeerId().SetFabricId(fabricInfo.GetFabricId()).SetNodeId(fabricInfo.GetNodeId()))
+                    .SetMac(FillMAC(mac))
+                    .SetPort(CHIP_PORT)
+                    .SetMRPRetryIntervals(CHIP_CONFIG_MRP_DEFAULT_ACTIVE_RETRY_INTERVAL,
```

Comment:
Technically, one of these should be `CHIP_CONFIG_MRP_DEFAULT_INITIAL_RETRY_INTERVAL`

Reply:
Happy to poke at this, but in a separate PR if that's OK @Damian-Nordic.  Was just trying to preserve behavior for the moment.

Reply:
Sure, I just spotted it in your diff, but I understand it's unrelated.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/8965